### ISSUE TITLE
ATAG 2.0 audit of CMS admin

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10515]
This is a tracking / umbrella issue in preparation for an [ATAG 2.0](https://www.w3.org/TR/ATAG20/) audit of our CMS admin interface. We’re expecting this audit to be our largest to date – covering as much of ATAG Part A and Part B as feasible. As far as I can tell, this would be the first ATAG 2.0 audit of a CMS ever! Or at least, the first publicly-shared that’d be longer than a [one-pager](https://www.w3.org/WAI/tools-list/authoring/).

For _Guideline A.1.1:_ in particular – the audit will include all of [WCAG _2.2_ AA-level](https://www.w3.org/TR/WCAG22/) (CR draft 3 as of now), and some of AAA-level guidelines as well as best practice recommendations as we see fit.

## Scope

- Wagtail admin views as defined via [UI mapping spreadsheet](https://docs.google.com/spreadsheets/d/1WaqARpHf99U0O94hypwHNjHA9yNpn4AnXnIT5AdJsm8/edit)
- Wagtail components as defined via [UI mapping spreadsheet](https://docs.google.com/spreadsheets/d/1WaqARpHf99U0O94hypwHNjHA9yNpn4AnXnIT5AdJsm8/edit)
- Wagtail CMS features with a definition TBD based on [bakerydemo website](https://github.com/wagtail/bakerydemo) features
- Representative sample selected based on [WCAG-EM](https://www.w3.org/WAI/test-evaluate/conformance/wcag-em/)

### WCAG

TBC. Options:

- [Living accessibility audit spreadsheet](https://docs.google.com/spreadsheets/d/1l7tnpEyJiC5BWE_JX0XCkknyrjxYA5T2aee5JgPnmi4/edit)
- Direct to GitHub issues
- [Be Inclusive](https://beinclusive.app/)
- <https://github.com/hidde/eleventy-wcag-reporter>
- OpenACR / VPAT

### ATAG

All of:

- [ATAG report tool](https://www.w3.org/WAI/atag/report-tool/)
- [WCAG authoring tools list entry](https://github.com/w3c/wai-authoring-tools-list)
- Bespoke template following ATAG 2.0 outline

### Automated WCAG testing

- Axe via Pa11y for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Axe via Accessibility Insights browser extension for non-default states for selected sample of views & states

### Semi-automated WCAG testing

Similar to past accessibility team reviews of Wagtail 3.0 and 4.0 via screenshots in Figma.

- Screenshot-taking & visual review in light mode for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Screenshot-taking & visual review in dark mode for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Screenshot-taking & visual review in forced-colors mode (WHCM) - light for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Screenshot-taking & visual review in forced-colors mode (WHCM) - dark for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Screenshot-taking & visual review for landmarks and headings for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))
- Screenshot-taking & visual review for accessible labels for 100% of scope views in their default state ([wagtail-tooling](https://github.com/thibaudcolas/wagtail-tooling))

### Manual WCAG testing

Across a representative sample of admin views,

- Screenshot-taking & visual review for tab order
- Manual keyboard navigation testing
- Manual touch navigation testing
- Manual 400% zoom testing
- Desktop screen readers: VoiceOver macOS, NVDA, JAWS
- Mobile screen readers: VoiceOver iOS, TalkBack
- Speech recognition: Dragon, Voice Control
- Magnification with Apple Zoom

### Manual ATAG auditing

Assessment of all ATAG guidelines feature-by-feature for the selected CMS functionality under review

- Matrix of ATAG guidelines applicability for all Wagtail features
- Where applicable, manual review of ATAG compliance considerations
- Passed/Failed/Cannot tell/Not applicable/Not checked for all features for all guidelines

**Comment by lb-** [comment]
Not sure if I should flag as a standalone issue but for the labels do not appear to be rendering for FieldBlock/widgets correctly. Discovered while reviewing #10705 

I have not dug into it further, but no `label` element appears to render at all.

<img width="772" alt="Screenshot 2023-08-05 at 8 32 36 am" src="https://github.com/wagtail/wagtail/assets/1396140/b26a572a-f31d-4cfb-b088-a4049a313324">

<img width="1746" alt="Screenshot 2023-08-05 at 8 27 41 am" src="https://github.com/wagtail/wagtail/assets/1396140/97a2c210-1095-4190-9df2-61c3b9283fe0">

Potential solution -> `client/src/components/StreamField/blocks/BaseSequenceBlock.js`
```diff
            <h2 class="w-panel__heading w-panel__heading--label" aria-level="3" id="${headingId}" data-panel-heading>
              <span data-panel-heading-text class="c-sf-block__title"></span>
-              <span class="c-sf-block__type">${blockTypeLabel}</span>
+              <label class="c-sf-block__type" for="??">${blockTypeLabel}</label>
```
We will then need to patch in the id when bound with the provided `idForLabel` or similar.

This seems to work in Firefox but not sure on a h2/h3 with `label` inside, also, not sure this will work for all contexts.

**Comment by thibaudcolas** [comment]
Thank you @lb-, this seems really solid as a separate issue so I’ve opened #11179 with your details.

**Comment by thibaudcolas** [comment]
This is now complete as of https://github.com/wagtail/rfcs/pull/89.
